main/src/library/Channel.flix

```diff
@@ -24,8 +22,8 @@ namespace Channel {
     ///
     pub def buffered(n: Int32): (Sender[t], Receiver[t]) \ IO =
         let m = if (n < 0) 0 else n;
-        let c = $CHANNEL_NEW$(m);
-        (c, c)
+        let c: Sender[t] = $CHANNEL_NEW$(m);
+        (c as Sender[t], c as Receiver[t])
```

Comment:
I am confused by the casts. Should this simply not give the right result without casts?

Reply:
Ah- its because the type rule should be updated.